### PR TITLE
Enable osin internal error logging

### PR DESCRIPTION
**Pull request by mrogers950**
osin now has an interface for internal error logging that provides some additional context on failure.

Example output:
```
=== RUN   TestClientCredentialFlow
--- PASS: TestClientCredentialFlow (0.00s)
=== RUN   TestAuthorizeStartFlow
I0206 14:58:22.818652   22392 access.go:561] osin: error=unauthorized_client, internal_error=<nil> get_client=client check failed, client_id=test
--- FAIL: TestAuthorizeStartFlow (0.00s)
osinserver_test.go:95: unexpected error: The client is not authorized to request a token using this method.
osinserver_test.go:135: unexpected error: Get http://127.0.0.1:39369/assert?code=KWhLRoflbPdPCNYNP6T3MpqZn4mv0zxKLU6TRg6QEAc&state=: EOF
```
@openshift/sig-security

### CODE REVIEW

File: pkg/oauthserver/osinserver/osinserver.go [hunk 1]
```diff
@@ -24,12 +25,23 @@ type Server struct {
 	errorHandler ErrorHandler
 }
 
+// Logger captures additional osin server errors
+type Logger struct {
+}
+
+func (l Logger) Printf(format string, v ...interface{}) {
+	if glog.V(3) {
```
[the remark this answers is not in the document]
Comment (php-coder) [reply pkg/oauthserver/osinserver/osinserver.go]: > What log level do we normally run online at?

It's a loglevel 2 AFAIR.

**Comment by mrogers950** [comment]
@enj I rebased and turned the V level down to 2, and changed InfoDepth to an ErrorDepth of 3.

File: pkg/oauthserver/osinserver/osinserver.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (enj) [reply pkg/oauthserver/osinserver/osinserver.go]: > It's a loglevel 2 AFAIR.

Confirm this with @jupierce

**Comment by enj** [comment]
/lgtm

@deads2k can you tag?

**Comment by deads2k** [comment]
So bumps require determination of whether you should actually bump the packages in question.  Separating the repo dependencies into yours, mine and ours buckets, we do not pin "yours" (kube's).  We always pin "mine" (but we have identified the set).  We sometimes pin "ours" depending on whether we really care what it is and who the other dependant is.

We are actively trying to make this better (see https://github.com/openshift/origin/pull/18543), but it isn't going to make 3.9.

**Comment by deads2k** [comment]
Oh, and syncing out from our vendor folder is currently in progress. Rules merged early this morning and @mfojtik is trying to enable the bot.

**Comment by mrogers950** [comment]
@deads2k I bumped only the osin dependency manually. I'm not sure how the vendor sync-out you mentioned affects this, so let me know if there is something I might need to do different here...

**Comment by deads2k** [comment]
> @deads2k I bumped only the osin dependency manually. I'm not sure how the vendor sync-out you mentioned affects this, so let me know if there is something I might need to do different here...

A manual change just leaves a time bomb for the next guy.  I know it's annoying, I know it can be tedious, I know it's not something you'd like to do.  I know this far more than the average person on the team.  Understanding how our product manages golang dependencies must not remain an arcane skill.  Please take the time to learn how to manage dependencies.

**Comment by mrogers950** [comment]
@deads2k after rebasing, this time running update-deps resulted in only updating osin (a "mine" now pinned at 2dc1b431) and k8s.io/kubernetes. PTAL

**Comment by mrogers950** [comment]
@deads2k nevermind, this dep update was wrong again.

**Comment by mrogers950** [comment]
@deads2k thanks for syncing kube! the dep bump is clean now. PTAL.

**Pull request merged**